modules/rostests/apitests/ntdll/NtAllocateVirtualMemory.c

```diff
@@ -600,7 +633,21 @@ CheckSomeDefaultAddresses(VOID)
                                      &Size,
                                      MEM_RESERVE | MEM_COMMIT,
                                      PAGE_READWRITE);
+    // Result varies between Windows versions.
     ok_ntstatus(Status, STATUS_SUCCESS);
+    if (NT_SUCCESS(Status))
+    {
+        trace("0x00000000(+) is available, as on ReactOS and Windows XP/S03/Vista_x32/7_x32\n");
+        // 0x00000000, 64k: Free.
+    }
+    else
+    {
+        trace("0x00000000(+) is not available, as on Windows 8.1_x32/Vista+_x64/Vista+_amd64\n");
+        ok_ntstatus(Status, STATUS_INVALID_PARAMETER_2);
+        skip("Following tests on first 64k region would fail too\n");
+        goto First64kIsUnavailable;
```

Comment:
If what you're testing here is "virtual address space layout on process start", why don't you make a table describing the expected layout and loop through the address space with NtQueryVirtualMemory? That seems much more straightforward than a ton of allocation calls.